components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2132,7 +2132,7 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
             workspaceType: mandatory(s.getWorkspaceType()) as WorkspaceType,
             workspaceClass: mandatory(s.getWorkspaceClass()),
             startTime: mandatory(s.getStartTime(), (t) => t!.toDate().toISOString()),
-            endTime: mandatory(s.getEndTime(), (t) => t!.toDate().toISOString()),
+            endTime: s.getEndTime()?.toDate().toISOString() || undefined,
```

Comment:
Do you need the `|| undefined`? . I think the optional chaining will ensure the expression has the value `undefined` when the end time is not set.

Reply:
That makes sense. Let me remove. I wasn't sure exactly how the `?` works.